gapir/cc/function_table.h

```diff
@@ -49,23 +55,23 @@ class FunctionTable {
   inline Function* lookup(Id id);
 
  private:
-  // Map of the supported function ids to the actual function implementations
-  std::unordered_map<Id, Function> mFunctions;
+  // Array of the actual function implementations by function ID
+  Function mFunctions[65536];
 };
 
 inline FunctionTable::Function* FunctionTable::lookup(Id id) {
-  auto func = mFunctions.find(id);
-  if (func == mFunctions.end()) {
+  Function& ret = mFunctions[(uint16_t)id];
+  if (ret == nullptr) {
     return nullptr;
   }
-  return &func->second;
+  return &ret;
 }
 
 inline void FunctionTable::insert(Id id, Function func) {
-  if (mFunctions.count(id) != 0) {
+  if (mFunctions[(uint16_t)id] != NULL) {
```

Comment:
What happens if id > 655536? Probably deserves an assert, as this will be a tricky bug to track down in the future./
Or make Id a 16-bit number.

Reply:
typedef uint16_t Id;

Reply:
Then the cast should not be necessary?

Reply:
And to make things more clear, we can change the constant 65536 to 
std::numeric_limits<Id>::max. That makes it perfectly clear that we cannot overflow.

Reply:
Fair.

Reply:
Can we also remove the cast (Also c-style casts are forbidden in the style guide)